### ISSUE TITLE
check_syntax: Target variable not in scope

### ISSUE THREAD

**Issue by ed-holland** [abaplint/abaplint#3358]
The table TSTOR is available in the dependency repository.
Even though this is old syntax, it is still valid.

Version used:
`node version:     v16.20.2`
`abaplint version: 2.110.7`

Message:
`./src/ecc/#jcs#fi.fugr.#jcs#fi_tstor_action.abap[306, 3]                     - Target variable tstor not found in scope (check_syntax) [E]`

Code:
![image](https://github.com/user-attachments/assets/f6c8b090-bb6b-466f-b389-9a6aae59f812)

**Comment by larshp** [comment]
where/how is `TSOTR` defined? can you help copy pasting the code example into the issue as text?

**Comment by ed-holland** [comment]
Tables definition at the beginning of the fuba.
![image](https://github.com/user-attachments/assets/f867f9cf-f15e-4771-8955-837241844b16)

**Comment by ed-holland** [comment]
Yes, inside the function scope.

**Comment by larshp** [comment]
heh

https://help.sap.com/doc/abapdocu_752_index_htm/7.52/en-us/abaptables.htm note "should only be declared in the global declaration part of a program"

reproduced as below

```abap
FORM foo.
  TABLES t100.
ENDFORM.

FORM bar.
  SELECT SINGLE * FROM t100.
ENDFORM.
```

**Comment by larshp** [comment]
I'll probably first add a rule to check for TABLES inside FORM or FUNCTION, as they are always global anyhow it doesnt make sense to declare them locally

**Comment by ed-holland** [comment]
This issue is also resolved with 2.111.0
